monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js

```diff
@@ -398,17 +399,43 @@ class ConfigurePageComponent extends AuthComponent {
   }
 
   sendCredentials() {
+    let islandCredentials = formatCredentialsForIsland(this.state.credentials.credentialsData);
+    let formattedCredentials = JSON.stringify(islandCredentials);
+    let formCredentialsIsland = {};
+    let credentialsDataId = {};
+    this.state.credentials.credentialsData.forEach((stateCredentials, stateCredentialsIndex) => {
+        formCredentialsIsland[stateCredentials.id] = formatCredentialsForIsland([stateCredentials]);
```

Comment:
Didn't we agree on modifying and only calling this function once per row, and storing that data to reuse here?